proxy/coco/src/peer.rs

```diff
@@ -702,6 +736,102 @@ mod test {
+
+        let alice_tmp_dir = tempfile::tempdir().expect("failed to create tempdir");
+        let alice_repo_path = alice_tmp_dir.path().join("radicle");
+        let config = config::default(alice_key.clone(), alice_tmp_dir.path())?;
+        let alice_peer = Api::new(config).await?;
+        let alice_peer_id = alice_peer.peer_id().clone();
+        let alice_addr = alice_peer.listen_addr();
+
+        let alice = alice_peer.init_owner(&alice_key, "alice")?;
+        let project = alice_peer.init_project(
+            &alice_key,
+            &alice,
+            &shia_le_pathbuf(alice_repo_path.clone()),
+        )?;
+        let urn = project.urn();
+
+        let bob_key = SecretKey::new();
+
+        let bob_tmp_dir = tempfile::tempdir().expect("failed to create tempdir");
+
+        let bob_config = config::default(bob_key.clone(), bob_tmp_dir.path())?;
+        let bob_peer = Api::new(bob_config).await?;
+        let _bob = bob_peer.init_owner(&bob_key, "bob")?;
+
+        let bobby = bob_peer.clone();
+        let url = urn.into_rad_url(alice_peer_id.clone());
+        let project_urn = tokio::task::spawn_blocking(move || {
+            bobby.clone_project(url, vec![alice_addr].into_iter())
+        })
+        .await
+        .expect("failed to join thread")?;
+
+        assert_eq!(
+            bob_peer
+                .list_projects()?
+                .into_iter()
+                .map(|project| project.urn())
+                .collect::<Vec<_>>(),
+            vec![project_urn.clone()]
+        );
+
+        let commit_id = {
+            let repo = git2::Repository::open(alice_repo_path.join(project.name()))?;
+            let oid = repo
+                .find_reference(&format!("refs/heads/{}", project.default_branch()))?
+                .target()
+                .expect("Missing first commit");
+            let commit = repo.find_commit(oid)?;
+            let commit_id = {
+                let empty_tree = {
+                    let mut index = repo.index()?;
+                    let oid = index.write_tree()?;
+                    repo.find_tree(oid)?
+                };
+
+                let author = git2::Signature::now(alice.name(), "alice@example.com")?;
+                repo.commit(
+                    Some(&format!("refs/heads/{}", project.default_branch())),
+                    &author,
+                    &author,
+                    "Successor commit",
+                    &empty_tree,
+                    &[&commit],
+                )?
+            };
+
+            let mut rad = repo.find_remote(config::RAD_REMOTE)?;
+            rad.push(&[&format!("refs/heads/{}", project.default_branch())], None)?;
+            commit_id
+        };
+
+        let bobby = bob_peer.clone();
+        let url = project_urn.into_rad_url(alice_peer_id.clone());
+        tokio::task::spawn_blocking(move || bobby.fetch(url, vec![alice_addr]))
+            .await
+            .expect("failed to join thread")?;
```

Comment:
```suggestion
            .expect("failed to fetch")?;
```

Reply:
Wellll actually, the error I'm expecting on is `JoinThread`. The `?` is the failure to fetch.

Reply:
Sneaky.